stellargraph/core/graph.py

```diff
@@ -1344,6 +1344,14 @@ def _edge_metrics_by_type_triple(self, metrics):
                 "weight": self._edges.weights,
             }
         )
+
+        # if graph is undirected and edges exist, we want to sort the triple
+        if not self.is_directed() and len(df):
+            sorted_types = df.apply(
+                lambda r: sorted([r.src_ty, r.tgt_ty]) + [r.rel_ty], axis=1
+            )
+            df[["src_ty", "tgt_ty", "rel_ty"]] = pd.DataFrame(sorted_types.tolist())
```

Comment:
Since this isn't touching the `rel_ty` column, I think this can be done with less Python iteration and less pandas.Series construction via NumPy:

```suggestion
            sorted_types = df[["src_ty", "tgt_ty"]].to_numpy()
            sorted_types.sort(axis=1)
            df[["src_ty", "tgt_ty"]] = sorted_types
```

This form might even allow removing the `len(df)` condition, although that's less important.